ui/module.go

```diff
@@ -0,0 +1,183 @@
+package ui
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	_ "modernc.org/sqlite" // SQLite driver
+
+	"github.com/GoCodeAlone/modular"
+)
+
+// UIModule provides web UI functionality for the workflow engine
+type UIModule struct {
+	name         string
+	config       UIConfig
+	server       *http.Server
+	dbService    *DatabaseService
+	authService  *AuthService
+	apiHandler   *APIHandler
+	logger       modular.Logger
+}
+
+// UIConfig represents the configuration for the UI module
+type UIConfig struct {
+	Address    string `yaml:"address" json:"address"`       // Server address (e.g., ":8080")
+	StaticDir  string `yaml:"staticDir" json:"staticDir"`   // Directory for static files
+	SecretKey  string `yaml:"secretKey" json:"secretKey"`   // JWT secret key
+	DatabaseDB string `yaml:"database" json:"database"`     // Database connection string
+}
+
+// NewUIModule creates a new UI module
+func NewUIModule(name string, config map[string]interface{}) *UIModule {
+	uiConfig := UIConfig{
+		Address:   ":8080",
+		StaticDir: "./ui/static",
+		SecretKey: "default-secret-key", // Should be overridden in production
+	}
+
+	// Parse configuration
+	if addr, ok := config["address"].(string); ok {
+		uiConfig.Address = addr
+	}
+	if staticDir, ok := config["staticDir"].(string); ok {
+		uiConfig.StaticDir = staticDir
+	}
+	if secretKey, ok := config["secretKey"].(string); ok {
+		uiConfig.SecretKey = secretKey
+	}
+	if dbPath, ok := config["database"].(string); ok {
+		uiConfig.DatabaseDB = dbPath
+	}
+
+	return &UIModule{
+		name:   name,
+		config: uiConfig,
+	}
+}
+
+// Name returns the module name
+func (m *UIModule) Name() string {
+	return m.name
+}
+
+// Dependencies returns the module dependencies
+func (m *UIModule) Dependencies() []string {
+	return []string{"database"}
+}
+
+// Configure sets up the UI module
+func (m *UIModule) Configure(app modular.Application) error {
+	return nil // Configuration is handled in Init
+}
+
+// Init initializes the UI module
+func (m *UIModule) Init(app modular.Application) error {
+	m.logger = app.Logger()
+
+	// Get database connection - try to get existing database service first
+	var db *sql.DB
+	if dbService := app.SvcRegistry()["database"]; dbService != nil {
+		if sqlDB, ok := dbService.(*sql.DB); ok {
+			db = sqlDB
+		}
+	}
+	
+	if db == nil {
+		// If no database service is available, create SQLite database
+		m.logger.Info("No database service found, creating SQLite database")
+		var err error
+		db, err = sql.Open("sqlite", "./workflow_ui.db")
+		if err != nil {
+			return fmt.Errorf("failed to open database: %w", err)
+		}
+	}
```

Comment:
If this module had a Constructor method, it could have the Database and other dependencies, dependency injected.